### ISSUE TITLE
[Question] Is it possible to get id from item in store?

### ISSUE THREAD

**Issue by swayf** [mobxjs/mobx-state-tree#643]
I found only this way, but getStateTreeNode is not exported

```javascript
if (isStateTreeNode(item)) {
     return getStateTreeNode(item).identifier;
}
```
is there some other way to do this? =)
Thank you in advance

**Comment by robinfehr** [comment]
could you describe your use case a bit further? :)

**Comment by swayf** [comment]
Let say I have a model with someMap: types.map(SomeType). Now I need to check in general way (I don't know in mixin what concrete SomeType I have if it is in the SomeMap). SomeMap.has need key to check it, but if I have only SomeType instance.

**Comment by robinfehr** [comment]
we’re currently working on a reflection api - it might help you as soon as it its out.

**Comment by robinfehr** [comment]
btw. there is a ‘getType’ function
it might help in your case
https://github.com/mobxjs/mobx-state-tree/blob/master/API.md#gettype

**Comment by swayf** [comment]
Sorry, my explanation was not good enough =)
In code will be better explained
```javascript
const Todo = types.model({
  id: types.identifier(types.number),
  task: types.string
})

const TodoStore = types.model({
  todos: types.map(Todo)
})

const todoItem = { task: "Grab cookie", id: 18 };
s.todos.put(todoItem) // put will infer key from the identifier

// here I need to now if todoItem in s.todos. in general way.. I can't do just
s.todos.has(todoItem.id) 
// because I should now that identifier for todoItem is 'id' field
// what I need is something like this, like 'put' function it infer key from items itself..
s.todos.has(todoItem) 
// or some general way to get key from todoItem myself, like
s.todos.has(Todo.getIdentifer(todoItem))

```

**Comment by mweststrate** [comment]
Ah, we should expose a getIdentifier(thing) utility, we already have that
internally

Op vr 9 feb. 2018 om 12:08 schreef Oleg Butovich <notifications@github.com>:

**Comment by mweststrate** [comment]
note that you can use map.values().contains(thing), but that is a lot more
inefficient


Op vr 9 feb. 2018 om 12:19 schreef Michel Weststrate <mweststrate@gmail.com

**Comment by mweststrate** [comment]
Closed in favor of #674.
